### PR TITLE
CSV Upload, Creating concepts with the given UUID.

### PR DESCRIPTION
**Pull request by mddubey**
During CSV Upload if the UUIDs are specified in the CSV the new concepts will be created with the given UUID.

[Jira Card](https://bahmni.atlassian.net/browse/BAH-403)

### CODE REVIEW

File: admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptSetPersisterIT.java [hunk 1]
```diff
@@ -157,4 +158,29 @@ public void should_fail_to_persist_if_conceptSetRow_introduces_cycle() throws Ex
...
+        Context.openSession();
+        Context.authenticate("admin", "test");
+        Concept persistedConcept = conceptService.getConceptByName(conceptRow.name);
+        assertNotNull(persistedConcept);
+        assertEquals(conceptRow.name, persistedConcept.getName(Context.getLocale()).getName());
+        assertEquals(uuid, persistedConcept.getUuid());
+        Context.flushSession();
+        Context.closeSession();
+    }
+
```
Comment (teleivo) [inline admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptSetPersisterIT.java]: can you please remove the newlines

File: admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptSetPersisterIT.java [hunk 2]
```diff
@@ -157,4 +158,29 @@ public void should_fail_to_persist_if_conceptSetRow_introduces_cycle() throws Ex
         Context.closeSession();
     }
 
+    @Test
+    public void shouldCreateNewConceptSetWithGivenUUID() throws Exception {
+        String uuid = UUID.randomUUID().toString();
+        ConceptSetRow conceptRow = new ConceptSetRow();
```
Comment (teleivo) [inline admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptSetPersisterIT.java]: can you please rename the `conceptRow` to `conceptSetRow`

File: reference-data/api/src/main/java/org/bahmni/module/referencedata/labconcepts/mapper/ConceptCommonMapper.java [hunk 3]
```diff
@@ -19,6 +19,8 @@
         org.openmrs.Concept openmrsConcept = new org.openmrs.Concept();
         if (conceptMetaData.getExistingConcept() != null) {
             openmrsConcept = conceptMetaData.getExistingConcept();
+        } else if (StringUtils.isNotBlank(conceptCommon.getUuid())){
```
Comment (teleivo) [inline reference-data/api/src/main/java/org/bahmni/module/referencedata/labconcepts/mapper/ConceptCommonMapper.java]: @angshu or @djazayeri I had trouble replicating the issue locally (explained it in jira), and my understanding of bahmni's packages/classes is still low, so maybe you can clarify this for me:

the above IT tests test Persisters from package org.bahmni.module.admin.csv.persister
which use Mapper's from package org.bahmni.module.admin.concepts.mapper and not referencedata/labconcepts

see

https://github.com/Bahmni/bahmni-core/blob/master/admin/src/main/java/org/bahmni/module/admin/csv/persister/ConceptPersister.java#L6

so I am wondering if the change here is in the right place. thanks!

File: admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptPersisterIT.java [hunk 4]
```diff
@@ -369,4 +370,30 @@ public void createNewConceptOfTypeNumericWithUnitsAndHinormalLownormal() throws
...
+        Concept persistedConcept = conceptService.getConceptByName(conceptRow.name);
+        assertNotNull(persistedConcept);
+        assertEquals(uuid, persistedConcept.getUuid());
+        assertEquals(conceptRow.name, persistedConcept.getName(Context.getLocale()).getName());
+        assertEquals(conceptRow.conceptClass, persistedConcept.getConceptClass().getName());
+        assertEquals("New concept description", persistedConcept.getDescription().getDescription());
+        assertEquals(0, persistedConcept.getSynonyms().size());
+        Context.flushSession();
+        Context.closeSession();
+
```
Comment (teleivo) [inline admin/src/test/java/org/bahmni/module/admin/csv/persister/ConceptPersisterIT.java]: remove newlines

File: reference-data/api/src/main/java/org/bahmni/module/referencedata/labconcepts/mapper/ConceptCommonMapper.java (hunk 3 above)
Comment (djazayeri) [reply reference-data/api/src/main/java/org/bahmni/module/referencedata/labconcepts/mapper/ConceptCommonMapper.java]: I'm not familiar with this code either, but from tracing through starting from ConceptPersister.persist this seems like the right change.

**Comment by teleivo** [comment]
@mddubey did you deploy bahmni with your changes and try to upload concept CSV from the admin section? I did and I do not see any change in behaviour. I am not sure if I am doing something wrong.

I uploaded a file like

concepts.csv
```
uuid,name,description,class,shortname,datatype,units,High Normal,Low Normal,locale,synonym.1,answer.1,answer.2,answer.3,reference-term-source,reference-term-code,reference-term-relationship
"d36bc411-3f10-11e4-adec-0800271c1b75","Bird",,"Misc","Bird","Numeric",,,,,,,,,,,
```

in the admin section, by selecting the radio button for concept like so

screen shot 2018-05-07 at 21 07 02

and it does get imported but with a different uuid, see

screen shot 2018-05-07 at 21 08 40

**Comment by mddubey** [comment]
Hey @teleivo,

Yes, I had deployed these changes in my local bahmni and I was using it for the purpose I had fixed it for. It will take some time for me to set up everything and verify from start. 

Can you just tell me you deployed these changes on top of what version of bahmni-installer? I will try with the same once I get sometime.

**Pull request merged**